content/docs/visualization/template_examples.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Template examples
+sort_rank: 4
+---
+
+# Template examples
+
+Prometheus has support for templating in both the summary/description field of
+alerts and also exposed as html endpoints. Templates have the ability to run
+queries against the databases, iterate over data, use conditionals, format data
+etc. This is based on the [go templating](http://golang.org/pkg/text/template/)
+system.
+
+## Simple alert field templates
+
+    ALERT InstanceDown
+      IF up == 0
+      FOR 5m
+      WITH {
+        severity="page"
+      }
+      SUMMARY "Instance {{$labels.instance}} down"
+      DESCRIPTION "{{$labels.instance}} of job {{$labels.job}} has been down for more than 5 minutes."
+
+
+Alert field templates will be executed for every rule iteration for every alert
+that fires, so keep any queries and templates lightweight. If you have a need for
+more complicated templates for alerts it is better to link to a console.
+
+## Simple iteration
+
+This displays a list hosts, and whether they are up:
+
+```
+{{ range query "up" }}
+  {{ .Labels.instance }} {{ .Labels.Value }}
+{{ end }}
+```
+
+`.` is given each sample in turn.
```

Comment:
"The special `.` variable contains the value of the current sample for each loop iteration."